File: Data Structures & Algorithms/missing-number/submission-2.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        
        m = max(nums)
        tot = m
        for i in range(m):
            tot += i
            tot -= nums[i]
        if tot in nums:
            tot = m + 1
        return tot 
        

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        List<Integer> starts = new ArrayList<>();
        Set<Integer> numSet = new HashSet<>();
        for(int num : nums) {
            numSet.add(num);
        }

        for(int num : nums) {
            if(!numSet.contains(num - 1)) {
                starts.add(num);
            }
        }
        int res = 0;
        for (int num : starts) {
            int count = 0;
            while (numSet.contains(num)){
                num += 1;
                count += 1;
            }
            res = Math.max(res, count);
        }

        return res;

    }
}
